File: workspace/src/image_receiver/src/image_receiver_node.cpp
#include <chrono>
#include <functional>
#include <memory>
#include <string>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
#include "sensor_msgs/msg/image.hpp"

class ImageReceiverNode : public rclcpp::Node
{
	public:
		ImageReceiverNode() : Node("image_receiver_node"), count_(0)
		{
			image_subscription = this->create_subscription<sensor_msgs::msg::Image>("image", 10, std::bind(&ImageReceiverNode::topic_callback, this, std::placeholders::_1));
		}
	private:
		void image_received_callback (const sensor_msgs::msg::Image::SharedPtr msg) const
		{
			RCLPP_INFO(this->get_logger(), "Image received!");
		}

		rclcpp::Subscription<sensor_msgs::msg::Image>::SharedPtr image_subscription;

};

int main(int argc, char* argv[])
{
	rclcpp::init(argc, argv);
	rclcpp::spin(std::make_shared<ImageReceiverNode>());
	rclcpp::shutdown();
	return 0;
}

File: workspace/src/py_image_receiver/py_image_receiver/image_receiver_node.py
import rclpy
from rclpy.node import Node
from sensor_msgs.msg import Image
import math

class ImageReceiverNode(Node):
    def __init__(self):
        super().__init__('image_receiver_node')
        self.image_subscription = self.create_subscription(Image, "image", self.image_received_callback, 10)
        self.clock_ = rclpy.clock.Clock()
        self.image_subscription

    def image_received_callback(self, msg):
        cur_time = self.clock_.now()
        secs = cur_time.nanoseconds//1000000000 - msg.header.stamp.sec
        nsecs = cur_time.nanoseconds%1000000000 - msg.header.stamp.nanosec
        self.get_logger().info(f"Image Received!\tTime diff: {secs}.{nsecs}\tseconds")

def main(args=None):
    rclpy.init(args=args)
    image_receiver_node = ImageReceiverNode()

    rclpy.spin(image_receiver_node)

    image_receiver_node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()
        
